node/agent.go

```diff
@@ -464,7 +470,7 @@ func updateInstance(state *servicestate.ServiceState, ctr *docker.Container) err
 // configureContainer creates and populates two structures, a docker client Config and a docker client HostConfig structure
 // that are used to create and start a container respectively. The information used to populate the structures is pulled from
 // the service, serviceState, and conn values that are passed into configureContainer.
-func configureContainer(a *HostAgent, client *ControlClient,
+func configureContainer(a *HostAgent, client ControlClientInterface,
```

Comment:
I just realized that `node.ControlClient` is itself an implementation of another interface - `dao.ControlPlane`. So instead of creating a new interface in node, you can change this line to be `client dao.ControlPlane` instead of `client *ControlClient`.  And then you can move the mock to `dao/mocks`